cmd/kube-apiserver/app/server.go

```diff
@@ -432,12 +460,12 @@ func BuildGenericConfig(s *options.ServerRunOptions, proxyTransport *http.Transp
 	kubeVersion := version.Get()
 	genericConfig.Version = &kubeVersion
 
-	storageFactory, err := BuildStorageFactory(s, genericConfig.MergedResourceConfig)
-	if err != nil {
-		return nil, nil, nil, nil, nil, err
+	storageFactory, lastErr := BuildStorageFactory(s, genericConfig.MergedResourceConfig)
+	if lastErr != nil {
+		return
 	}
-	if err := s.Etcd.ApplyWithStorageFactoryTo(storageFactory, genericConfig); err != nil {
-		return nil, nil, nil, nil, nil, err
+	if lastErr = s.Etcd.ApplyWithStorageFactoryTo(storageFactory, genericConfig); lastErr != nil {
```

Comment:
Without all these lastErr refactorings, the PR would be much much smaller. Would feel better with a minimal patch for the release branch.

Reply:
ok, will do some cleanup

Reply:
We add a new return value for `CreateKubeAPIServerConfig`, so all the return lines have to be changed even without lastErr

Reply:
Right. Hmm. For easier pickability, we should maybe leave the commit complete then.